edx_event_bus_kafka/internal/consumer.py

```diff
@@ -146,10 +146,11 @@ def reset_offsets(consumer, partitions):
             if offset_timestamp is None:
                 return
 
+            full_topic = get_full_topic(self.topic)
             # Get the offset from the epoch in seconds.
             offset_timestamp_s = int(offset_timestamp.timestamp())
             # We construct partitions with the epoch timestamp in the offset position.
-            timestamp_partitions = [TopicPartition(self.topic, p.partition, offset_timestamp_s) for p in partitions]
+            timestamp_partitions = [TopicPartition(full_topic, p.partition, offset_timestamp_s) for p in partitions]
```

Comment:
Can we get a test that fails on the former and passes on the update, to ensure we won't have a regression?

Reply:
It's nearly impossible to test this deep into the code because of all the kafka library stuff around it.

Reply:
You'd have to do your best with mocking. Some ideas:
1. Mock consumer.subscribe and get the function sent to on_assign (e.g. reset_offsets).
2. Call this function, and check that a mocked consumer.offsets_for_times gets called using full topic names.
3. Ensure that mocked consumer.assign is called and passed the mocked result of consumer.offsets_for_times.